### ISSUE TITLE
copybuffer - inscript '^O' is not working

### ISSUE THREAD

**Issue by rjshrjndrn** [robbyrussell/oh-my-zsh#6442]
I had to manually use `bindkey '^o' copybuffer` in my .zshrc

**Comment by mpauper** [comment]
I am having the exact same issue, only have the following plugins.

plugins=(git zsh-syntax-highlighting copybuffer)

Context: PopOS 21.04 & Kitty terminal emulator

**Comment by mcornella** [comment]
> I am having the exact same issue, only have the following plugins.
> 
> plugins=(git zsh-syntax-highlighting copybuffer)
> 
> Context: PopOS 21.04 & Kitty terminal emulator

What does `bindkey '^O'` show when it isn't working? Also, are you sure you have a command written in the command line? It doesn't copy the last command, but what you have written at the moment.

**Comment by mpauper** [comment]
> What does `bindkey '^O'` show when it isn't working? Also, are you sure you have a command written in the command line? It doesn't copy the last command, but what you have written at the moment.

Yes, I have a command typed and then I try pressing `Ctrl + o` before hitting enter to run the command. 

`bindkey '^O'` outputs `"^O" self-insert`
I am not sure what that means.

**Comment by mcornella** [comment]
I can't reproduce your issue, can you run `zsh -xvic exit &> ~/omz-debug.log` and post the generated file?

**Comment by mpauper** [comment]
Thank you for looking into it! Here is the file:

[omz-debug.log](https://github.com/ohmyzsh/ohmyzsh/files/7144152/omz-debug.log)

**Comment by mcornella** [comment]
That doesn't help narrow it down, it looks like the plugin is running successfully. I'd bet that it has something to do with `zsh-syntax-highlighting`, although it works on my system. Can you disable it and try if the key binding is properly set up afterwards?

**Comment by mpauper** [comment]
:( Nope, it still won't work. In fact, I removed all the other plugins except for copybuffer, but I still get `^O`

**Comment by mcornella** [comment]
OK try the following:

1. Add the lines below at the top of your .zshrc file:
   ```zsh
   exec 3>&2 2> ~/omz-debug.log
   ```
2. Run `zsh -xv`.
3. When you've tried to trigger copybuffer and it doesn't work, close the zsh session with `exit`.
4. Remove the line added in step 1.
5. Post the generated file.

**Comment by mpauper** [comment]
There it is:
[omz-debug.log](https://github.com/ohmyzsh/ohmyzsh/files/7144478/omz-debug.log)

To clarify, after trying `Ctrl + o` and getting `^O` appended to my written command, I used `Ctrl + c` to cancel and then `exit`.

**Comment by mcornella** [comment]
Oh so simple:
```
# Enable vim mode
bindkey -v
+/home/<user>/.zshrc:106> bindkey -v
```

The bindkey is defined for emacs mode, I'll push a fix 😬

**Comment by mcornella** [comment]
The fix is pushed, try `omz update` and it should be working correctly.

**Comment by mpauper** [comment]
Wonderful, it works! Thank you! 

I have no idea what you just did and I have already used up a lot of your time, but if you feel like giving me an ELI5 go ahead :D

**Comment by mcornella** [comment]
`bindkey '^O' copybuffer` means define a key binding for CTRL+O to run the `copybuffer` function, **for the current keymap**.

When OMZ starts, it sets the current keymap to `emacs` (with `bindkey -e`), so all the bindkey commands set the key bindings for the emacs keymap. If you later change to another keymap, like `viins` (`bindkey -v`), those key bindings aren't defined for it.

The fix involved just defining the key binding for the rest of the normally used keymaps with the `-M <keymap>` argument passed to `bindkey`.

**Comment by mpauper** [comment]
Oh, understood!

Thanks again, have a great weekend!

-------- Original Message --------
